test/Tests/src/Data/Pair_Spec.enso

```diff
@@ -11,13 +12,90 @@ spec = Test.group "Pair" <|
     Test.specify "should allow mapping" <|
         Pair.new 1 2 . map x->x+1 . should_equal (Pair.Value 2 3)
 
-    Test.specify "should act as a length 2 Vector" <|
+    Test.specify "should have length 2" <|
         Pair.new "A" "B" . length . should_equal 2
+
+    Test.specify "should allow reversing" <|
+        Pair.new 1 2 . reverse . should_equal (Pair.Value 2 1)
+
+    Test.specify "should have allow getting by index" <|
```

Comment:
Please turn this `Pair_Spec` suite into one parametrized by the _"pair constructor"_ and then test the suite on `Pair.new`, `Cons first (Cons second Nil)`, vector & co. to make sure all the implementations behave exactly the same when they hold two values.

Reply:
Good idea - have amended the tests to be like this.

Reply:
I really like this idea and I think in the future it would be really cool to expand this into a `Sequence_Collection_Spec` for testing arbitrary length sequences, we could have at least some shared tests for `Vector` and `List`, maybe some (but not all) for `Range` too. Definitely not something to do right now though.